repose-aggregator/components/services/datastore-service/datastore-service-impl/src/main/java/org/openrepose/core/services/datastore/impl/distributed/CacheRequest.java

```diff
@@ -65,34 +59,16 @@ private static String getCacheKey(HttpServletRequest request) {
         return cacheKey;
     }
 
-    private static String getHostKey(HttpServletRequest request) {
-        final String hostKeyHeader = request.getHeader(DatastoreHeader.HOST_KEY);
-
-        if (StringUtilities.isBlank(hostKeyHeader)) {
-            throw new MalformedCacheRequestException(MalformedCacheRequestError.NO_DD_HOST_KEY);
-        }
-
-        return hostKeyHeader;
-    }
-
     public static boolean isCacheRequestValid(HttpServletRequest request) {
         return request.getRequestURI().startsWith(CACHE_URI_PATH);
     }
 
     public static String urlFor(InetSocketAddress remoteEndpoint, String key, boolean useHttps) {
-        String proto = "http";
-        if (useHttps) {
-            proto = "https";
-        }
-        return new StringBuilder(proto).append("://").append(remoteEndpoint.getAddress().getHostAddress()).append(":").append(remoteEndpoint.getPort()).append(CACHE_URI_PATH).append(key).toString();
+        return "http" + (useHttps ? "s" : "") + "://" + remoteEndpoint.getAddress().getHostAddress() + ":" + remoteEndpoint.getPort() + CACHE_URI_PATH + key;
```

Comment:
Other than the proto/useHttps change, the two lines should result in the same bytecode. The only difference is what we want to see in the code.